src/core/api_client_parallel.py

```
@@ -37,80 +36,98 @@ async def _get_with_key(self, session: aiohttp.ClientSession, key_index: int,
+                            status=response.status,
+                            message=f"Rate limit hit on key {key_index}"
+                        )
+
                     response.raise_for_status()
                     return await response.json(), key_index
-                    
-            except Exception as e:
+
+            except asyncio.TimeoutError:
+                self.error_counts[key_index] += 1
+                raise aiohttp.ClientTimeout(f"Request timeout on key {key_index}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)